src/main.rs

```diff
@@ -266,14 +269,27 @@ const APP: () = {
             .FULL_UPDATE
             .lock(|fu| core::mem::replace(&mut *fu, false));
         if full_update {
-            resources.DISPLAY.set_full();
+            resources
+                .DISPLAY
+                .set_lut(&mut *resources.SPI, Some(RefreshLUT::FULL))
+                .unwrap(); //resources.DISPLAY.set_full();
         }
         resources
             .DISPLAY
-            .set_display(&mut *resources.SPI, &display)
+            .update_frame(&mut *resources.SPI, &display.buffer())
+            .unwrap();
+        resources
+            .DISPLAY
+            .display_frame(&mut *resources.SPI)
             .unwrap();
-        resources.DISPLAY.update(&mut *resources.SPI).unwrap();
-        resources.DISPLAY.set_partial();
+        while resources.DISPLAY.is_busy() {}
```

Comment:
The busy wait should be in the if, allowing the microcontroller to sleep during the rendering